src/Features/Core/Portable/ExternalAccess/Watch/Api/WatchHotReloadService.cs

```diff
@@ -55,17 +54,18 @@ public Update(Guid moduleId, ImmutableArray<byte> ilDelta, ImmutableArray<byte>
             (_, _, _) => ValueTaskFactory.FromResult(ImmutableArray<ActiveStatementSpan>.Empty);
 
         private readonly IEditAndContinueWorkspaceService _encService;
+        private readonly Task<ImmutableArray<string>> _getCapabilitiesTask;
 
-        public WatchHotReloadService(HostWorkspaceServices services)
-            => _encService = services.GetRequiredService<IEditAndContinueWorkspaceService>();
+        public WatchHotReloadService(HostWorkspaceServices services, Task<ImmutableArray<string>> getCapabiltiesTask)
```

Comment:
Why a task? Looking at the other PR it doesn't seem like there is any reason this can't just be an `ImmutableArray<string>`. Is awaiting it expensive?

Reply:
I was able to change this..